pipeline.ts

```diff
@@ -48,13 +45,14 @@ export function createRedisPipeline(
   }
 
   async function send(cmds: string[]): Promise<RawReplyOrError[]> {
+    const writerAsBuffer = connection.writer! as BufWriter;
     const msg = cmds.join("");
-    await writer.write(encoder.encode(msg));
-    await writer.flush();
+    await writerAsBuffer.write(encoder.encode(msg));
+    await writerAsBuffer.flush();
     const ret: RawReplyOrError[] = [];
     for (let i = 0; i < cmds.length; i++) {
       try {
-        const rep = await readReply(reader);
+        const rep = await readReply(connection.reader! as BufReader);
```

Comment:
I think this type assertion is unnecessary.
```suggestion
        const rep = await readReply(connection.reader!);
```